### PR TITLE
Move addAnnotation and removeAnnotation to ClassifierInstance

### LINKED ISSUE #153: Annotations on non-dynamic nodes
Currently, it is only possible to add annotations on dynamic nodes because the _annotations_ collection is an immutable view and the _addAnnotation_ method is only available on _AbstractClassifierInstance_ objects.

### PR DESCRIPTION
**Pull request by ftomassetti**
Move to ClassifierInstance annotations specific methods previously only available on AbstractClassifierInstance
Fix #153

### CODE REVIEW

File: core/src/main/java/io/lionweb/lioncore/java/model/ClassifierInstance.java [hunk 1]
```diff
@@ -1,14 +1,36 @@
...
 
+  /**
+   * Given a specific Annotation type it returns either the list of instances of that Annotation
+   * associated to the Node.
+   */
+  @Nonnull
+  List<AnnotationInstance> getAnnotations(@Nonnull Annotation annotation);
+
+  /**
+   * If an annotation instance was already associated under the Annotation link used by this
```
Comment (enikao) [inline core/src/main/java/io/lionweb/lioncore/java/model/ClassifierInstance.java:23]: I thought we removed `Annotation.multiple` from M3? So we can always attach several instances of the same annotation.

Comment (ftomassetti) [reply core/src/main/java/io/lionweb/lioncore/java/model/ClassifierInstance.java]: You are right. The logic was already like you suggested. I have now updated also the comment.

**Comment by enikao** [comment]
Why not moving it to `AbstractClassifierInstance`? Annotations can have annotations.

**Comment by ftomassetti** [comment]
> Why not moving it to AbstractClassifierInstance? Annotations can have annotations.

We are moving them to `ClassifierInstance,` which is more general than `AbstractClassifierInstance`. `AnnotationInstance extends `ClassifierInstance`, so the methods are also available there.

**Comment by ftomassetti** [comment]
I think we covered the comments, so unless there are more doubts I would go ahead and merge this one

**Pull request merged**